src/main/java/games/strategy/engine/framework/startup/ui/ClientSetupPanel.java

```diff
@@ -173,114 +172,92 @@ public boolean canGameStart() {
           final JButton button = new JButton(m_dontTakeAction);
           button.setMargin(buttonInsets);
-          m_playerComponent = button;
+          playerComponent = button;
         } else {
-          m_playerComponent = new JLabel("");
+          playerComponent = new JLabel("");
         }
       }
       setWidgetActivation(disableable);
     }
 
     private void setWidgetActivation(final boolean disableable) {
-      m_playerNameLabel.setEnabled(m_enabledCheckBox.isSelected());
-      m_playerLabel.setEnabled(m_enabledCheckBox.isSelected());
-      m_playerComponent.setEnabled(m_enabledCheckBox.isSelected());
-      m_alliance.setEnabled(m_enabledCheckBox.isSelected());
-      m_enabledCheckBox.setEnabled(disableable);
+      playerNameLabel.setEnabled(enabledCheckBox.isSelected());
+      playerLabel.setEnabled(enabledCheckBox.isSelected());
+      playerComponent.setEnabled(enabledCheckBox.isSelected());
+      alliance.setEnabled(enabledCheckBox.isSelected());
+      enabledCheckBox.setEnabled(disableable);
     }
 
     public boolean isPlaying() {
-      return m_playerLabel.getText().equals(clientModel.getMessenger().getLocalNode().getName());
+      return playerLabel.getText().equals(clientModel.getMessenger().getLocalNode().getName());
     }
 
     public JComponent getPlayerComponent() {
-      return m_playerComponent;
+      return playerComponent;
     }
 
     public String getLocalType() {
-      return m_localPlayerType;
+      return localPlayerType;
     }
 
-    private final Action m_takeAction = new AbstractAction("Play") {
-      private static final long serialVersionUID = 9086754428763609790L;
-
-      @Override
-      public void actionPerformed(final ActionEvent e) {
-        clientModel.takePlayer(m_playerNameLabel.getText());
-      }
-    };
-    private final Action m_dontTakeAction = new AbstractAction("Dont Play") {
-      private static final long serialVersionUID = 8735891444454338978L;
-
-      @Override
-      public void actionPerformed(final ActionEvent e) {
-        clientModel.releasePlayer(m_playerNameLabel.getText());
-      }
-    };
-    private final ActionListener m_disablePlayerActionListener = new ActionListener() {
-      @Override
-      public void actionPerformed(final ActionEvent e) {
-        if (m_enabledCheckBox.isSelected()) {
-          clientModel.enablePlayer(m_playerNameLabel.getText());
-        } else {
-          clientModel.disablePlayer(m_playerNameLabel.getText());
-        }
-        setWidgetActivation(true);
+    private final Action m_takeAction =
+        SwingAction.of("Play", e -> clientModel.takePlayer(getName().getText()));
+    private final Action m_dontTakeAction =
+        SwingAction.of("Dont Play", e -> clientModel.releasePlayer(getName().getText()));
+    private final ActionListener m_disablePlayerActionListener = e -> {
+      if (enabledCheckBox.isSelected()) {
+        clientModel.enablePlayer(getName().getText());
```

Comment:
For some weird reason the compiler complains `playerNameLabel` might not have been initilized although it has to be...